pkg/pgmodel/extension.go

```diff
@@ -270,6 +276,13 @@ func getNewExtensionVersion(extName string,
 		if availableVersions[i].GT(defaultVersion) {
 			continue
 		}
+
+		// if upgradePreRelease is false skip the pre-releases.
+		// if the current installed version is an rc version return it as an valid available version.
+		if len(availableVersions[i].Pre) > 0 && !upgradePreRelease && availableVersions[i].String() != currentVersion.String() {
```

Comment:
I would log a warning here like we do in the major version update check to let people know a pre-release is available but we won't be updating to it.

Reply:
But this logs multiple times If there are multiple `rc` versions available.